src/tests/JIT/opt/And/IntAnd.cs

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace CodeGenTests
+{
+    class IntAnd
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void SideEffect()
+        {
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool Test_UInt32_UInt32_And(uint x, uint y)
+        {
+            // X64-NOT: movzx
+
+            // We expect 'and reg8, reg8'.
+            // X64: and {{[a-z]+[l|b]}}, {{[a-z]+[l|b]}}
+
+            if ((byte)(x & y) == 0)
+            {
+                SideEffect();
+                return true;
+            }
+            return false;
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static bool Test_UInt32_UInt32_CastByte_And(uint x, uint y)
```

Comment:
Do you need a `CastByte` variant for the RHS of the operation to make sure that both tree shapes work?

Reply:
Not sure what you mean here. What is the RHS in this case?

Reply:
`if ((byte)(x & (byte)y) == 0)`

Sorry, RHS was inaccurate.. "right operand of the &" would have been better

Reply:
I see, I think it's fine and we don't need to add another test case. In this test case, if the second op of `and` is an 8-bit register, I can feel pretty confident that casting the byte will still result in using an 8-bit register.

Reply:
The next test case below this one casts both ops and still gives an 8-bit register.